script.js

```diff
@@ -0,0 +1,310 @@
+
+      const ruKeyLower = document.createElement('span');
+      ruKeyLower.classList.add('on');
+      ruKeyLower.innerHTML = this.ruLowerCase[i];
+      this.wrapper.lastChild.firstChild.append(ruKeyLower);
+
+      const engKeyUpper = document.createElement('span');
+      engKeyUpper.classList.add('off');
+      engKeyUpper.innerHTML = this.engUpperCase[i];
+      this.wrapper.lastChild.lastChild.append(engKeyUpper);
+
+      const engKeyLower = document.createElement('span');
+      engKeyLower.classList.add('on');
+      engKeyLower.innerHTML = this.engLowerCase[i];
+      this.wrapper.lastChild.lastChild.append(engKeyLower);
+
+      if (i === 14 || i === 29 || i === 42 || i === 55) {
+        key.classList.add('float-none');
+      }
+    }
+  }
+
+  checkKey(str) {
+    switch (str) {
+      case 'Enter':
+        this.textArea.value += '\n';
+        break;
+      case 'Tab':
+        this.textArea.value += '\t';
+        break;
+      case 'Backspace':
+        this.deleteSymbolPrev();
+        break;
+      case 'Delete':
+        this.deleteSymbolNext();
+        break;
+      case 'Del':
+        this.deleteSymbolNext();
+        break;
+      case 'ArrowLeft':
+        this.arrowMove('ArrowLeft');
+        break;
+      case 'ArrowRight':
+        this.arrowMove('ArrowRight');
+        break;
+      case 'ArrowUp':
+        this.arrowMove('ArrowUp');
+        break;
+      case 'ArrowDown':
+        this.arrowMove('ArrowDown');
+        break;
+      case 'CapsLock':
+        this.changeCase();
+        break;
+      default:
+    }
+  }
+
+  mouseClick(e) {
+    const target = e.target.closest('.key');
+    if (target == null) {
+      return;
+    }
+    const targetKey = target.className.split(' ')[1];
+    this.checkKey(targetKey);
+    target.classList.add('active');
+    setTimeout(() => {
+      target.classList.remove('active');
+    }, 500);
+    const text = e.target.closest('.key').querySelector('.activeKeys').querySelector('.on').textContent;
+    if (text.length > 1) {
+      return;
+    }
+    this.textArea.value += text;
+  }
+
+  keyDown(e) {
+    e.preventDefault();
+    if (e.code === 'CapsLock') {
+      this.changeCase();
+      return;
+    }
+    if (e.code === 'ShiftLeft' || e.code === 'ShiftRight') {
+      this.caseUpper();
+    }
+    this.checkKey(e.code);
+    const key = document.querySelector(`.${e.code}`);
+    if (key === null) {
+      return;
+    }
+    key.classList.add('activeKeyBoard');
+    const text = document.querySelector(`.${e.code}`).querySelector('.activeKeys').querySelector('.on').textContent;
+    if (text.length > 1) {
+      return;
+    }
+    this.textArea.value += text;
+  }
+
+  keyUp(e) {
+    if (e.code === 'CapsLock') {
+      return;
+    }
+    const key = document.querySelector(`.${e.code}`);
+
+    if (key === null) {
+      return;
+    }
+
+    key.classList.remove('activeKeyBoard');
+    if (e.key === 'Shift') {
+      if (this.capsMode === true) {
```

Comment:
Use truthy and falsy values instead of obvious comparison with boolean type.

Reply:
Fixed